File: INTRODUCCION-POO/ejercicio3/Producto.java
package ejercicio3;

// Diseña una clase Producto que tenga atributos nombre, precio y
// stock. Agrega métodos para vender productos y reabastecer el stock.
// a) Define la clase y los atributos.
// b) Agrega un método vender(cantidad: Int) que reste del stock
// c) Agrega un método reabastecer(cantidad: Int) que sume al stock.

public class Producto {
    // atributos
    private String nombre;
    private double precio;
    private int stock;

    // constructor
    public Producto(String nombre, double precio, int stock) {
        this.nombre = nombre;
        this.precio = precio;
        this.stock = stock;
    }

    // metodos
    public void vender(int cantidad){
        if(cantidad <= stock){
            this.stock -=  cantidad;
            System.out.println("Se vendioron " + cantidad + " unidades de " + nombre);
            System.out.println("Stock actual: " + this.stock);
        }
        else{
            System.out.println("Error: No hay suficiente stock para vender " + cantidad + " unidades de " + nombre);
        }
    }

    public void reabastecer(int cantidad){
        this.stock += cantidad;
        System.out.println("Se agregaron " + cantidad + " unidades de " + nombre);
        System.out.println("Stock actual: " + this.stock);
    }

    //mostrar detalles del producto
    public void  mostrarProducto(){
        System.out.println("Nombre: " + nombre);
        System.out.println("Precio: Bs. " + precio);
        System.out.println("Stock disponible: " + stock);
    }
}

File: INTRODUCCION-POO/ejercicio3/producto.py
class Producto:
	def __init__(self, nombre: str, precio: float, stock: int):
		self.nombre = nombre
		self.precio = precio
		self.stock = stock

	def vender(self, cantidad: int):
		if cantidad > self.stock:
			print(f"No hay suficiente stock para vender {cantidad} unidades de {self.nombre}.")
		else:
			self.stock -= cantidad
			print(f"Se vendieron {cantidad} unidades de {self.nombre}. Stock restante: {self.stock}")

	def reabastecer(self, cantidad: int):
		self.stock += cantidad
		print(f"Se reabastecieron {cantidad} unidades de {self.nombre}. Stock actual: {self.stock}")

# programa principal

p1 = Producto("Galletas", 2.5, 10)
p1.vender(3)
p1.reabastecer(5)
p1.vender(15)
p1.vender(4)
